src/common/type-check.ts

```diff
@@ -5,7 +5,7 @@ import { Evaluable, Result } from './evaluable'
  * @param value Tested value.
  */
 export function isNumber(value: Result): value is number {
-  return typeof value === 'number' && isFinite(value)
```

Comment:
Is this potentially a change in contract?

Reply:
I thought about it, but I don't think so. 
* Infinity didn't seem to be actively supported. There were no tests for it and not documented.
* Although it's possible to express Infinity in an illogical condition created in a JS array, it will not work while being transported through JSON (and potentially other holders).
![image](https://github.com/user-attachments/assets/56f68833-80f6-42a4-afa5-f5612497ae81)
Seems that it gets converted to `null` instead of throwing error
* Also, before this implementation I don't really see how Infinity would be generated in runtime

Maybe I need to validate if it's possible that the simplify would render Infinity, which would be a problem

Reply:
Interesting, `comparison` simplify was never expecting that a single operand could be simplified. I should code this, which would then expose the problem above. I can use the isFinite check to prevent the issue.